mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoImplementationGenerator.java

```diff
@@ -417,6 +417,7 @@ protected ClassName getPrincipalTypeName() {
       String fieldName = entry.getValue();
       // - create an instance
       initAsyncBuilder.addStatement("$1T $2L = new $1T(context)", fieldTypeName, fieldName);
+      initAsyncBuilder.addStatement("throwIfKeyspaceMissing($L, context.getSession())", fieldName);
```

Comment:
One thought is that in theory, a user could only specify Dao methods that don't require a keyspace, like `@GetEntity` and `@SetEntity`, we could choose not to fail in this case, but not sure if it's worth worrying about?

Reply:
We could move the check to each entity helper method, we have the context there so I think we have everything we need. Another advantage of that is that the check would also kick in when you call a helper method from a query provider.

We can introduce an `EntityHelperBase` parent class to factor the method.

Reply:
yeah this seems easy enough to do.  I had initially went about it this way, but seemed like validating in each of these methods was less simple than validating it on entity helper creation. I'll change it.

Reply:
Adding `EntityHelperBase` also eliminates some boilerplate in the generated code, so that looks like a good change overall.